src/three/objects/artwork.ts

```diff
@@ -0,0 +1,122 @@
+    }
+};
+
+export class Artwork3DObject extends Mesh {
+    public isHighlighted: boolean = false;    
+    private _artworkRef: DimensionizedCdaItem;
+    public userData: ArtworkUserData;
+    public readonly name = "artwork";
+    // private _cachedSize: THREE.Vector3 | null = null;
+
+    private constructor (geometry: BufferGeometry, material: MeshBasicMaterial, artwork: DimensionizedCdaItem) {
+        super(geometry, material);
+        this._artworkRef = artwork;
+        this.userData = {
+            year: artwork.dating.begin,
+            rawItem: artwork
+        }
+        this.scale.set(artworkProperties.scale, artworkProperties.scale, artworkProperties.scale);
+        this.geometry.computeBoundingBox();
+    }
+
+    static async buildArtworkObject(artwork: DimensionizedCdaItem): Promise<Artwork3DObject> {
+        const geometry = createGeometry(artwork.parsedDimensions);
+        const url = dataProxyUrl(artwork.images.overall.images[0].sizes.medium.src);
+        const texture = await new THREE.TextureLoader().loadAsync(url);
+        const material = new THREE.MeshBasicMaterial({ 
+            color: 0xFFFFFF,
+            map: texture
+        });
+        return new this(geometry, material, artwork);
+    }
+
+    // TODO: I really don't know why this calculation fails. It will result in 1/2 
+    //       the actual object size. I guess its some weird Shenannigan with mutable data structure
+    //       and getter methods. Leaving it for now as it is as I need to focus on other things.
+    // get size(): THREE.Vector3 {
+    //     if(this._cachedSize !== null) {
+    //         return this._cachedSize;
+    //     }
+    //     if(this.geometry.boundingBox === null) {
+    //         this.geometry.computeBoundingBox();
+    //     }
+    //     const box = this.geometry.boundingBox!;
+    //     this._cachedSize = box.max.sub(box.min).multiply(this.scale).multiplyScalar(0.1);
+    //     return this._cachedSize;
+    // }
+
+    get width(): number {
+        if(this.geometry.boundingBox === null) {
+            this.geometry.computeBoundingBox();
+        }
+        const box = this.geometry.boundingBox!;
+        return (box.max.x - box.min.x) * this.scale.x * 0.1;
+        // return this.size.x;
+    }
+
+    get height(): number {
+        if(this.geometry.boundingBox === null) {
+            this.geometry.computeBoundingBox();
+        }
+        const box = this.geometry.boundingBox!;
+        return (box.max.y - box.min.y) * this.scale.y * 0.1;
+        // return this.size.y;
+    }
+
+    get depth(): number {
+        if(this.geometry.boundingBox === null) {
+            this.geometry.computeBoundingBox();
+        }
+        const box = this.geometry.boundingBox!;
+        return (box.max.z - box.min.z) * this.scale.z * 0.1;
+        // return this.size.z;
+    }
+
+    highlight(): void {
```

Comment:
Highlighting ist really cool to clarify what you get the metadata for, but it kind of takes the free view to the 3D picture, because it gets all greenish. Maybe an outline highlight would be cool in the future.